### DISCUSSION TITLE
Background Image Won't Display on GitHub Pages

### DISCUSSION THREAD

**Discussion by nathan-friedman-designs** [community/community#35195] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi all,

I've published my portfolio using GitHub Pages, and all the code seems to work except that my background image won't display. That image (portBackground.jpg) is the lynchpin of my portfolio, because it sets the stage, mood, and story of the site's design. Following similar discussions I discovered on the internet, I've changed the image from a .png to a .jpg, moved it into an exclusive "images" folder, and I even cleared my browser's cache from the previous hour before I uploaded it. So it's bothering me to no end as to why the image won't display. Would you please help me out?

My GitHub Pages published site link: https://nathan-friedman-designs.github.io/nathanfriedmanportfolio/

My repository: https://github.com/nathan-friedman-designs/nathanfriedmanportfolio

Thanks in advance!

**Comment by antonblr** [comment, marked as answer]
You need to change the image url from  `url("images\portBackground.jpg");` to `"../images/portBackground.jpg"`
* backslash to forward slash
* one level up (`../`), since `images/` dir is outside of the `css` dir

https://github.com/nathan-friedman-designs/nathanfriedmanportfolio/blob/420f4f5ea4aeccc41ba89e4212eb818f77136e34/css/index.css#L6

**Reply by nathan-friedman-designs** [reply]
Yes, thank you, it worked!

**Reply by m3c-ode** [reply]
But then, locally, it won't be working anymore. How do you adapt with this?

**Reply by sliyarli** [reply]
same problem here, did you find any solution?

**Reply by DarkOphanim** [reply]
Works on local too! Make sure everything is correct. 
_example_ 
This is not gonna work => background-image: url(/images/myImage.png); or
                                            background-image: url(images/myImage.png); or
                                            background-image: url(./images/myImage.png); or
                                            background-image: url(../images/myImage.png); or
This will work (local or online) => background-image: url("../images/myImage.png");

**Reply by 5Mayuri5** [reply]
omg it worked thank you so much!!

**Reply by tamanna-hash** [reply]
I have only few hours to submit my assignment and you saved me...thanks

**Comment by jfullstackdev** [comment]
seems it's already fixed by now ? 

![image](https://user-images.githubusercontent.com/47092464/194090603-19661cba-9b0d-4daf-8767-20a29d053933.png)

**Comment by JoeMille** [comment]
cheers for posting this it took me ages to work that out

**Comment by anjylo** [comment]
Also had this problem but my solution was removing the quote from the path.

e.g
background: url(path/to/file/)

**Comment by JoeMille** [comment]
Remove the .. from the file paths as that directs them somewhere else so if a file path is ../assets/images/image.jpg then remove the “..” so it would be /assets/images/image.jpgOn 31 Aug 2023, at 17:37, Şahin Əliyarlı ***@***.***> wrote:﻿
same problem here, did you find any solution?

**Comment by sliyarli** [comment]
> But then, locally, it won't be working anymore. How do you adapt with this?



> Remove the .. from the file paths as that directs them somewhere else so if a file path is ../assets/images/image.jpg then remove the “..” so it would be /assets/images/image.jpgOn 31 Aug 2023, at 17:37, Şahin Əliyarlı ***@***.***> wrote:﻿ same problem here, did you find any solution? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

i tried these solutions but none of them worked:

`  background-image: url("../images/pattern-background-desktop.svg");`
`  background-image: url("./images/pattern-background-desktop.svg");`
`  background-image: url("/images/pattern-background-desktop.svg");`

**Comment by JoeMille** [comment]
Sorry I spoke too soon, reading back over my CSS file from my last submission I removed the “quotes” inside URL so it would be- url(../images/pattern-background-desktop.svg)After doing this it fixed the problem of my images not displaying when page deployed. Hope this helps!Sent from my iPhoneOn 31 Aug 2023, at 18:49, Şahin Əliyarlı ***@***.***> wrote:﻿

But then, locally, it won't be working anymore. How do you adapt with this?


Remove the .. from the file paths as that directs them somewhere else so if a file path is ../assets/images/image.jpg then remove the “..” so it would be /assets/images/image.jpgOn 31 Aug 2023, at 17:37, Şahin Əliyarlı @.> wrote:﻿ same problem here, did you find any solution? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: @.>

i tried these solutions but none of them worked:
  background-image: url("");
  background-image: url("./images/pattern-background-desktop.svg");
  background-image: url("/images/pattern-background-desktop.svg");

**Reply by sliyarli** [reply]
thanks for the help

**Comment by mondoakim** [comment]
Hi guys am faced with the same problem my background image wont display in the download file. What am I missing? Please help! My css code is below.
.background{
width:100%;
height:100vh;
background-image: url(/img/bottles.jpg);
background-repeat: no-repeat;
background-size: cover;
background-position: fixed

**Reply by sliyarli** [reply]
change it to background-image: url(../../img/bottles.jpg);

just keep typing "../" until you select the image this way and you're gonna be fine

**Reply by mondoakim** [reply]
Hi thank you very much for your response and help well appreciated. I now have another problem I cant get my carousel slideshow to work even after typing the right code and CDN links. What am I missing? Here is the code below I look forward to your response.
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Tap & Bottles</title>

<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-4bw+/aepP/YC94hEpVNVgiZdgIC5+VKNBQNGCHeKRQN+PtmoHDEXuppvnDJzQIu9" crossorigin="anonymous">

<script src="https://kit.fontawesome.com/ae75e7a7d4.js" crossorigin="anonymous"></script>

<link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.css"/>

<link rel="stylesheet" href="CSS/tapandbottles.css">
</head>
<body>
  
  <header>
    <div class="container">
      <div class="row">
      <div class="col-md-4 col-12 col-12">
      <div class="btn-group">
        <button class="btn dropdown-toggle my-md-4 my-2 text-white" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">GBP</button>
        <div class="dropdown-menu">
          <a href="#" class="dropdown-item">ERU-Euro</a>
        </div>
      </div>
      </div>
      <div class="col-md-4 col-12 text center">
        <h2 class="my-md-3 site title text-white">Tap & Bottles</h2>
      </div>
      <div class="col-md-4 col-12 text-right">
        <p class="my-md-4 header-links">
          <a href="#" class="px 2">Sign In</a>
          <a href="#" class="px 1">Create An Account</a>
        </p>
      </div>
      </div>
    </div>
    <div class="container-fluid p-0">
      <nav class="navbar navbar-expand-lg bg-body-tertiary bg-white">
        <div class="container-fluid">
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
          </button>
          <div class="collapse navbar-collapse" id="navbarNav">
            <ul class="navbar-nav">
              <li class="nav-item">
                <a class="nav-link active" aria-current="page" href="#">HOME</a>
              </li>
              <li class="nav-item">
                <a class="nav-link" href="#">FEATURES</a>
              </li>
              <li class="nav-item">
                <a class="nav-link" href="#">COLLECTION</a>
              </li>
              <li class="nav-item">
                <a class="nav-link" href="#">SHOP</a>
              </li>
              <li class="nav-item">
                <a class="nav-link" href="#">ABOUT US</a>
              </li>
            </ul>
          </div>
          <div class="navbar-nav">
            <li class="nav-item border rounded-circle m2-search-icon">
              <i class="fa-solid fa-magnifying-glass p-2"></i>
            </li>
            <li class="nav-item border rounded-circle m2-basket-icon">
              <i class="fa-solid fa-shopping-basket p-2"></i>
            </li>
          </div>
        </div>
      </nav>
    </div>
  </header>
  
  <main>

    
    <div class="container-fluid p-0">
      <div class="site-slider">
    <div class="slider-one">
    <div>
      <img src="/img/shopify.webp" class="img-fluid" alt="banner 1" width="500" height="250">
    </div>
    <div>
      <img src="/img/shopify2.webp" class="img-fluid"   alt="banner 2" width="500" height="250">
    </div>
    <div>
      <img src="image2/beer glass.jpg" class="img-fluid"   alt="banner 2" width="500" height="250">
    </div>
    </div>
      </div>
    </div>
    

  </main>

  
  <footer></footer>

    


  <script src="/JS/tapandbottles.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/js/bootstrap.bundle.min.js" integrity="sha384-HwwvtgBNo3bZJJLYd8oVXjrBZt8cqVSpeBNS5n7C8IVInixGAoxmnlMuBnhbgrkm" crossorigin="anonymous"></script>
  <script type="text/javascript" src="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.min.js"></script>
</script>
</body>
</html>

**Reply by mondoakim** [reply]
/** First Slider */
$(".slider-one").slick();

**Comment by Diogo010101** [comment]
I tested all the ways and they all worked, but you need a few minutes for them to work on the Git-Hub pages.

https://diogo010101.github.io/projeto_cafeteria/

**Comment by paocorrales** [comment]
Hi, 

I have the same issue but none of the possible solutions worked for me. The images are where they are supposed to be but I keep getting a 404. I deployed the web to netflify and it works there. 

Github pages> https://rse-r.github.io/intro-programacion/
Netflify> https://intro-programacion.netlify.app/
Repository> https://github.com/rse-r/intro-programacion

The issue is with the icons inside each box.

**Reply by mostafa16dev** [reply]
Hi,

check the path of images, for me it was `../images/bg-card-front.png` and I changed it to `./images/bg-card-front.png` to point to the current directory of the project before it was referencing the root directory

**Comment by jonhnmedeiros** [comment]
Use a trick that u can use style atribute in html file with background-image and url ('./images/description-background.jpg'). This work 100% for me on GitHub Pages.

`            <div class="hero-mission" style="background-image: url('./resources/images/img-mission-background.jpg')">
                <div class="hero-text">
                    <h2>Our Mission</h2>
                    <h4>Handpicked, Artisanally Curated, Free Range, Sustainable, Small Batch, Fair Trade, Organic Tea
                    </h4>
                </div>
            </div>`

https://github.com/jonhnmedeiros/tea_cosy

https://jonathanmedeiros.dev/tea_cosy/

**Reply by githuvlad** [reply]
У меня тожк не чего не работает все варианты пробовал решений так и не нашлось через  background-image: добовлять нету смысла он в git page не работает
